src/App.tsx

```diff
@@ -24,22 +23,24 @@ function App() {
   });
 
   useEffect(() => {
-    setIsLogedin(accessToken ? true : true);
-    console.log("isLogedin", isLogedin, accessToken);
-  }, []);
-
-  const [userProfile, setuserProfile] = useState<UserModal | null>();
+    setIsLogedin(accessToken ? true : false);
+  }, [accessToken]);
 
+  const [userName, setuserName] = useState<string | null>(
+    window.localStorage.getItem("userName")
+      ? window.localStorage.getItem("userName")
+      : ""
```

Comment:
Keep this logic separate for readability.

Reply:
Also this should be `token`

Reply:
Updated